etcdctl/ctlv3/command/role_command.go

```diff
@@ -170,11 +170,30 @@ func roleGrantPermissionCommandFunc(cmd *cobra.Command, args []string) {
 		ExitWithError(ExitBadArgs, err)
 	}
 
-	rangeEnd, rerr := rangeEndFromPermFlags(args[2:])
-	if rerr != nil {
-		ExitWithError(ExitBadArgs, rerr)
+	key := args[2]
```

Comment:
dedup code into shared function?

```go
key, rangeEnd := permRange(args[2:])
```

Reply:
ok, I'll dedup them. Thanks for the comment.